exporter/datadogexporter/internal/translator/dimensions.go

```diff
@@ -0,0 +1,97 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/translator/utils"
+)
+
+const (
+	dimensionSeparator = string(byte(0))
+)
+
+type metricsDimensions struct {
+	name string
+	tags []string
+	host string
+}
+
+// getTags maps an attributeMap into a slice of Datadog tags
+func getTags(labels pdata.AttributeMap) []string {
+	tags := make([]string, 0, labels.Len())
+	labels.Range(func(key string, value pdata.AttributeValue) bool {
+		v := value.AsString()
+		tags = append(tags, utils.FormatKeyValueTag(key, v))
+		return true
+	})
+	return tags
+}
+
+// AddTags to metrics dimensions. The argument may be modified.
+func (m *metricsDimensions) AddTags(tags ...string) metricsDimensions {
+	return metricsDimensions{
+		name: m.name,
+		// append the field to the passed argument,
+		// so that the slice we modify is the one we get as an argument.
+		tags: append(tags, m.tags...),
```

Comment:
Should we just defensively copy `tags` here? I feel like even if it's documented we will eventually forget somewhere

Reply:
I agree that we should copy `tags`, it feels weird to modify the argument of a function that doesn't seem like it should.

Reply:
Done